src/main/java/com/drajer/bsa/ehr/service/impl/EhrFhirR4QueryServiceImpl.java

```diff
@@ -60,7 +68,12 @@
   public HashMap<ResourceType, Set<Resource>> getFilteredData(
       KarProcessingData kd, HashMap<String, ResourceType> resTypes) {
 
-    authorizationService.getAuthorizationToken(kd);
+    String secret = kd.getHealthcareSetting().getClientSecret();
+    if (secret == null || secret.isEmpty() || secret.isBlank()) {
```

Comment:
`.isBlank` was introduced in Java 11, so if we stick to Java 8 we'll need to change that. One possibility is to use StringUtils, which seems to already be included via some dependency:
`import org.apache.commons.lang3.StringUtils;`

```suggestion
    if (StringUtils.isNotBlank(secret)) {
```

(or alternatively, just drop the `isBlank` and use a trim() before isEmpty, or even just drop the blank check altogether, probably not a big deal )